### DISCUSSION TITLE
More detail on how to use the sd card

### DISCUSSION THREAD

**Discussion by stefmsft** [roleoroleo/sonoff-hack#41] [Q&A]
Hi, that's sound stupid but I ve tried all the steps ( downloaded the latest release, formated the sd with fat32, copied the boot.sh and the directory at the root) but the cam still use to standard firmware. I can see it on ewelink and a request to its IP on 8080 give nothing. Its like my sd card was completly ignored. What I did wrong ? Should I do a factory reset before using the procedure you describe ?
But advance thanks if you can help
Cheers

**Comment by roleoroleo** [comment]
A reset is not needed.
Simply decompress the release archive into the sd card.
If it doesn't work, please try another sd card.
If you are a Windows user, use 7zip to decompress the file.

**Reply by GL640** [reply]
i did the same and got the same problem. nothing happens. camera operates as normal and no http acces

maybe its because i used a 128gb sdcard? do you really need a les then 16gb card?

**Comment by GL640** [comment]
i did the same
maybe its because i used a 128gb sdcard? do you really need a les then 16gb card?

**Comment by roleoroleo** [comment]
I don't know if the cam supports a 128GB card.
An old 1-2GB card is enhough.

**Reply by GL640** [reply]
in micro sd i dont have or can not easy find that low gb card. i do have tons of normal size 1-8gb cards but that dont fit.
i ave fitted a 128gb card and that works with the factory video recording also the specs state that 128 is the max size.
but i use a 128 gb and formatted it on windows and also formatted it on the sonoff cam itself using the app . the devices records motion but when i put the files on it for the hack so the boot file in root and the hack folde in rood nothing happens when i reboot the device.
i do have the correct version of firmware installed and i have the correct model B firmware used.\

would really like to know if a less then 16gb is reallyy need before i go and buy a low gb card like that for nothing.
the intructions state les then 16gb so i geuss thats there not for nothging?

**Comment by roleoroleo** [comment]
What model is your cam?

**Reply by GL640** [reply]
as stated in my last comment "i do have the correct version of firmware installed and i have the correct model B firmware used.\"

its a model b not c or do i need to find more details info in the app?

**Reply by roleoroleo** [reply]
If your cam uses the sd card to save videos the sd card should work and the hack should work.
Probably the problem is different.

What tool did you use to decompress the archive?
If you are a Windows user, please try to use 7zip (for example WinRar could be a problem).

**Reply by GL640** [reply]
i have used winzip25-lan 
i can try to use the other one but the file are exstracted and it are just files and folders with the correct name. dont understand wy a different program would make any difference. unless file properties are incorrecly unpacked?

i wil try 7zip, could take a while before i get round to it.
and report back

**Reply by roleoroleo** [reply]
For example WinRar changes the newline format from linux to windows.
And shell scripts do not work.

**Reply by GL640** [reply]
OK! using 7zip and it worked!
only thing i now discover is that when i when to PTZ the page with the PTZ does not show the image and or the camera reboots itself
using chrome and firefox

**Reply by roleoroleo** [reply]
Enable swap file.

**Reply by GL640** [reply]
okthx but i prever not to swap to mutch to the sd card it would fail sooner i think
i also wathc the camera using a 3e party app and now with onvif i have ptz control.

i do notice the cam seems to crash en reboot a lot also what i thinks is more disturbing is that it always goes to the same viewing angle ( PTZ position ) and this is poiting the wrong way i now have to adjust this several times.
can i set a default PTZ location for when it boots?

**Reply by GL640** [reply]
do i need to edit 
in system.sh or can i do this from the web interface ( set a default point? ) 
insmod /mnt/mtd/ipc//app/drive/ptz_drv.ko factory="Links" AutoRun=1 Horizontal=3500 Vertical=900

**Reply by GL640** [reply]
EDIT i already see it in the web interface sorry.

**Comment by jezzaaa** [comment]
FYI, I used a 32GB card and partitioned it into 2x16GB volumes, and it worked just fine.

I'm wondering if I can use the second partition to store images, or maybe mount it as /opt and install packages.

**Comment by drticcz** [comment]
I just used 128GB card and it seems to work fine. One big partition per the card. Is there something I should be aware of? Should I reformat using 16GB boot partition and the rest for the data?

```
[root@GK]# df -h
Filesystem                Size      Used Available Use% Mounted on
/dev/root                 3.0M      3.0M         0 100% /
tmpfs                    17.9M         0     17.9M   0% /dev
tmpfs                    17.9M     44.0K     17.8M   0% /tmp
tmpfs                    17.9M    192.0K     17.7M   1% /var
tmpfs                    17.9M         0     17.9M   0% /bin
tmpfs                    17.9M     76.0K     17.8M   0% /usr
tmpfs                    17.9M         0     17.9M   0% /sbin
tmpfs                    48.0M      8.0K     48.0M   0% /dev/shm
/dev/mtdblock3          576.0K    280.0K    296.0K  49% /mnt/mtd
/dev/mtdblock4            2.1M      2.1M         0 100% /mnt/mtd/ipc/app
tmpfs                    17.9M    424.0K     17.4M   2% /etc
tmpfs                    17.9M     44.0K     17.8M   0% /mnt/mtd/ipc/app/res/En/di.wav
tmpfs                    17.9M     44.0K     17.8M   0% /mnt/mtd/ipc/app/res/En/Internet_connected_Welcome_to_use_cloud_camera.wav
tmpfs                    17.9M     44.0K     17.8M   0% /mnt/mtd/ipc/app/res/En/WiFi_connect_success.wav
tmpfs                    17.9M     44.0K     17.8M   0% /mnt/mtd/ipc/app/script/dhcp.sh
tmpfs                    17.9M     44.0K     17.8M   0% /mnt/mtd/ipc/app/colink
/dev/mmcblk0p1          119.1G    106.9M    118.9G   0% /mnt/mmc
/dev/mmcblk0p1          119.1G    106.9M    118.9G   0% /mnt/mmc/sonoff-hack/www/alarm_record
```

**Reply by roleoroleo** [reply]
Sometimes there is a problem if the sd card is slow and requires time to mount the partition.
But if it works fine, a big partition is not a problem.
